contracts/integrations/test/utils/function_assertions.ts

```diff
@@ -0,0 +1,68 @@
+import { PromiseWithTransactionHash } from '@0x/base-contract';
+import { TransactionReceiptWithDecodedLogs } from 'ethereum-types';
+
+export interface ContractGetterFunction {
+    callAsync: (...args: any[]) => Promise<any>;
+}
+
+export interface ContractWrapperFunction extends ContractGetterFunction {
+    awaitTransactionSuccessAsync?: (...args: any[]) => PromiseWithTransactionHash<TransactionReceiptWithDecodedLogs>;
+}
+
+export interface Result {
+    data?: any;
+    success: boolean;
+    receipt?: TransactionReceiptWithDecodedLogs;
+}
+
+export interface Condition {
+    before: (...args: any[]) => Promise<any>;
+    after: (beforeInfo: any, result: Result, ...args: any[]) => Promise<any>;
```

Comment:
It might be over-engineering to pass the call args into `after`. It also makes these callbacks unnecessarily unwieldy. If we need the call args, the `before` function could just return an object that includes them.

Reply:
All of the assertions that I've written so far do actually make use of these arguments. This said, I understand your point and I think that it's valid. 

I'm debating whether to make this change now or "wait and see" until we've actually made more use of the framework. I'm not convinced that it would actually be nicer to have to pass the arguments to `after` since I've always used them.